js/main.js

```diff
@@ -1,58 +1,106 @@
-/* eslint-disable no-use-before-define */
-// Create array of books
-const bookTitle = document.querySelector('#title');
-const bookAuthor = document.querySelector('#author');
-const bookAdd = document.querySelector('#available-books');
-const btn = document.querySelector('#add-button');
-
-if (!localStorage.getItem('bookInfo')) {
-  localStorage.setItem('bookInfo', JSON.stringify([]));
+/* eslint-disable max-classes-per-file */
+// ======== book class: book representation =========
+class Book {
+  constructor(title, author) {
+    this.title = title;
+    this.author = author;
+  }
 }
 
-let books;
+// ======= class storage: handle local storage ==========
 
-function saveBooks(book) {
-  localStorage.setItem('bookInfo', JSON.stringify(book));
-}
+class Store {
+  static getBooks() {
+    let books;
+    if (localStorage.getItem('books') === null) {
+      books = [];
+    } else {
+      books = JSON.parse(localStorage.getItem('books'));
+    }
+    return books;
+  }
+
+  static bookAdd(book) {
+    const books = Store.getBooks();
+    books.push(book);
+    localStorage.setItem('books', JSON.stringify(books));
+  }
 
-/* eslint-disable no-use-before-define */
+  static bookRemoval(del) {
+    const books = Store.getBooks();
+    books.forEach((book, index) => {
+      if (book.del === del) {
+        books.splice(index, 1);
+      }
+    });
 
-function displayBookData() {
-  books = JSON.parse(localStorage.getItem('bookInfo'));
-  userUpdate();
+    localStorage.setItem('books', JSON.stringify(books));
+  }
 }
-displayBookData();
 
-function removeBook() {
-  books.splice(this, 1);
-  saveBooks(books);
-  displayBookData();
-}
+// ======== UI class: handle ui ========
 
-function userUpdate() {
-  bookAdd.innerHTML = '';
-  books.forEach((data, index) => {
-    const newBook = document.createElement('div');
-    newBook.classList.add('new-book');
-    const para = document.createElement('p');
-    para.textContent = `${data.bookTitle} By ${data.bookAuthor}`;
-    const removebtn = document.createElement('button');
-    removebtn.textContent = 'Remove';
-    removebtn.addEventListener('click', removeBook.bind(index));
-    newBook.appendChild(para);
-    newBook.appendChild(removebtn);
-    bookAdd.appendChild(newBook);
-  });
+class UI {
+  static displayBooks() {
+    const books = Store.getBooks();
+    books.forEach((book) => UI.addBookToList(book));
+  }
+
+  static addBookToList(book) {
```

Comment:
- [ ] Adding book process are not stored correctly. When a book is added and the page is refreshed, the added book is not shown on the book list. This indicates that data is not properly being saved into the local storage. Please be reminded that for this project, it is required that refactoring your code need not break the functionality of the app. Kindly ensure to improve upon this aspect